File: coding-test/programmers/src/kr/co/programmers/practice/SumBetweenTwoIntegers.java
package kr.co.programmers.practice;

public class SumBetweenTwoIntegers {

    public static void main(String[] args) {
        int a = 3;
        int b = 5;
        System.out.println(new Solution().solution(a, b));
    }

    static class Solution {
        public long solution(int a, int b) {
            return (long)(Math.abs(b - a) + 1) * (a + b) / 2;
        }
    }
}

File: coding-test/javascript/src/practice/sum-between-two-integers.js
// @ts-check

/**
 *
 * @param {number} a
 * @param {number} b
 * @returns {number}
 */
function solution(a, b) {
  if (a > b) {
    let temp = a
    a = b
    b = temp
  }

  let answer = 0
  for (let i = a; i <= b; i += 1) {
    answer += i
  }

  return answer
}

function diffSolution(a, b) {
  return ((a + b) * (Math.abs(b - a) + 1)) / 2
}

const a = 3
const b = 5

console.log(solution(a, b))
console.log(diffSolution(a, b))
